lib/builder/builder.go

```diff
@@ -513,20 +508,21 @@ func (b *Builder) collectUpgradeDependencies() (result *libapp.Dependencies, err
 		return nil, trace.Wrap(err)
 	}
 	result = &libapp.Dependencies{}
-	for _, version := range b.UpgradeVia {
+	for _, runtimeVersion := range b.UpgradeVia {
+		app, err := apps.GetApp(loc.Runtime.WithVersion(runtimeVersion))
```

Comment:
Cannot use the host application to avoid pulling in and marking unrelated packages are part of this (`runtimeVersion`) intermediate step.